### PR TITLE
Allow modified views to fill their parent if a child requires it

### PR DESCRIPTION
**Pull request by carson-katri**
For instance, a `ModifiedContent` view wrapping a `VStack` with a `Spacer` will allow the stack to fill the parent's height by putting `height: 100%;` on itself. This goes for all views that have a child `SpacerContainer` that fills either axis.

### CODE REVIEW

**Comment by ie-ahm-robox** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakDOM/DOMRenderer.swift#L81** - Line should be 100 characters or less: currently 127 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakDOM/Shapes/Shape.swift#L21** - Line should be 100 characters or less: currently 125 characters (`line_length`)
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 8ab706a3ded072d71e97d3134f8f4fc6eab3d340
</p>

File: Sources/TokamakDOM/Modifiers/LayoutModifiers.swift [hunk 1]
```diff
@@ -79,6 +79,7 @@ private extension EdgeInsets {
 }
 
 extension _PaddingLayout: DOMViewModifier {
+  public var orderDependent: Bool { true }
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Modifiers/LayoutModifiers.swift:82]: nit

```suggestion
  public var isOrderDependent: Bool { true }
```

File: Sources/TokamakDOM/Modifiers/ViewModifier.swift [hunk 2]
```diff
@@ -19,6 +19,12 @@ public typealias ModifiedContent = TokamakCore.ModifiedContent
 
 public protocol DOMViewModifier {
   var attributes: [String: String] { get }
+  /// Can the modifier be flattened?
+  var orderDependent: Bool { get }
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Modifiers/ViewModifier.swift:23]:
```suggestion
  var isOrderDependent: Bool { get }
```

File: Sources/TokamakDOM/Modifiers/ViewModifier.swift [hunk 3]
```diff
@@ -19,6 +19,12 @@ public typealias ModifiedContent = TokamakCore.ModifiedContent
...
+}
+
+extension DOMViewModifier {
+  public var orderDependent: Bool { false }
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Modifiers/ViewModifier.swift:27]:
```suggestion
  public var isOrderDependent: Bool { false }
```

File: Sources/TokamakDOM/Modifiers/ViewModifier.swift [hunk 4]
```diff
@@ -42,6 +48,7 @@ extension _ZIndexModifier: DOMViewModifier {
 }
 
 extension _BackgroundModifier: DOMViewModifier where Background == Color {
+  public var orderDependent: Bool { true }
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Modifiers/ViewModifier.swift:51]:
```suggestion
  public var isOrderDependent: Bool { true }
```

File: Sources/TokamakDOM/Modifiers/_ViewModifier_Content.swift [hunk 5]
```diff
@@ -15,16 +15,43 @@
...
+}
+
 extension ModifiedContent: ViewDeferredToRenderer where Content: View {
   public var deferredBody: AnyView {
     if let domModifier = modifier as? DOMViewModifier {
-      return AnyView(HTML("div", domModifier.attributes) {
-        content
-      })
+      if let adjacentModifier = content as? AnyModifiedContent,
+        !(adjacentModifier.anyModifier.orderDependent || domModifier.orderDependent) {
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Modifiers/_ViewModifier_Content.swift:37]:
```suggestion
        !(adjacentModifier.anyModifier.isOrderDependent || domModifier.isOrderDependent) {
```

File: Sources/TokamakDOM/Shapes/Shape.swift [hunk 6]
```diff
@@ -32,6 +32,7 @@ extension _OverlayModifier: DOMViewModifier where Overlay == _ShapeView<_Stroked
 
 // TODO: Implement arbitrary clip paths with CSS `clip-path`
 extension _ClipEffect: DOMViewModifier {
+  public var orderDependent: Bool { true }
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/Shapes/Shape.swift:35]:
```suggestion
  public var isOrderDependent: Bool { true }
```

**Pull request merged**